### DISCUSSION TITLE
What can we do for Kepler to reach 1.0?

### DISCUSSION THREAD

**Discussion by jgomezselles** [sustainable-computing-io/kepler#1217] [General]
Hi there!

I hope it’s ok to open this discussion here. Please let me know if you think I should bring it somewhere else.

I was reading a [blog post](https://www.cncf.io/blog/2023/10/11/the-road-to-scaphandre-v1-0-challenges-and-improvements-to-come-on-it-energy-consumption-evaluation/) in the CNCF blog page. It adds some lines towards what's needed for project Scaphandre to reach 1.0 version.

Thinking about Kepler, I'd like to ask this community about what do you think it's needed to reach 1.0 here.

In case it helps the discussion:

- I've been looking at the [semver definition](https://semver.org/#how-do-i-know-when-to-release-100):  _“If your software is being used in production, it should probably already be 1.0.0. If you have a stable API on which users have come to depend, you should be 1.0.0. If you’re worrying a lot about backward compatibility, you should probably already be 1.0.0.”_

- Looking at CNCF’s [graduated](https://www.cncf.io/projects/) info: _“Graduated and incubating projects are considered stable and are used successfully in production environments.”_

- Kepler is now cataloged as “[Sandbox](https://www.cncf.io/sandbox-projects/)”.
Of course the fact that a project is in a Sandbox state does not imply it’s not stable nor production ready.

I’d like to raise this discussion and see what’s the community take here, and how could anyone contribute to reach the status. Based on the previous "definitions", the following questions could help here:

- Is Kepler “ready for production”?
    - What would be the meaning for that?
    - Under which use cases is it ready? Which ones not?

- Is it stable?
    - What would be the meaning for that?
    - Under which use cases is it stable? Which ones not?
    - Would it make sense to add "experimental" flags to certain features? For example, metrics exposed from RAPL -> Stable, metrics for VMs -> Experimental, or which HW, VM setups...
    - Would it make sense to have a compatibility matrix like [this one](https://hubblo-org.github.io/scaphandre-documentation/compatibility.html) and add the "maturity level" there?
  
- Is there a roadmap for 1.0.0?
    - Should there be one?
    - Is it too soon?
    - How can others contribute here?

After the discussion, maybe we can help there! I hope this makes sense.
Thanks!

**Comment by rootfs** [comment]
@jgomezselles thanks for starting this. We'll get the action items into release planning

**Comment by LJTian** [comment]
Also concerned.
